### PR TITLE
Add per-conversation streaming switch (deliveryMode='stream')

### PR DESCRIPTION
**Pull request by pranavjoshi001**
**Description**:
This change introduces support for the deliveryMode property when posting activities through the Direct Line SDK.

**Background**

ABS introduces streaming responses when clients send activities with deliveryMode='stream'. DirectLineJS previously had no way to set deliveryMode, blocking adoption.

**Changes in this PR**

- Added optional deliveryMode field to the base IActivity interface (covers message, event, typing).
- Added DirectLineOptions.streaming boolean. When true, every outgoing activity is forced to deliveryMode='stream' (overriding any preset value).
- Left behavior unchanged when streaming is false/omitted: deliveryMode is not sent (ABS treats it as 'normal').
- Updated postActivity to inject deliveryMode.
- Added tests coverage

**Backward Compatibility**

- Existing integrations unaffected; field is optional and only sent when streaming=true.

**Delivery mode doc** - https://github.com/microsoft/Agents/blob/main/specs/activity/protocol-activity.md#delivery-mode

### CODE REVIEW

File: src/directLine.test.ts [hunk 1]
```diff
@@ -243,4 +243,146 @@ describe('MockSuite', () => {
...
+            const actual: Array<DirectLineExport.Activity> = [];
+            subscriptions.push(directline.activity$.subscribe(a => actual.push(a)));
+
+            scheduler.flush();
+
+            expect(presetActivity.deliveryMode).toStrictEqual('normal');
+            expect(actual[0].deliveryMode).toStrictEqual('normal');
+        });
+
+        test('Streaming + 403 post returns retry and keeps deliveryMode=stream on activity', () => {
```
Comment (compulim) [inline src/directLine.test.ts]: This is nice corner case.

Consider test permutation so it will test everything you tested above. 2x the power.

File: src/directLine.test.ts [hunk 2]
```diff
@@ -243,4 +243,146 @@ describe('MockSuite', () => {
...
+            };
+
+            subscriptions.push(lazyConcat(scenario()).observeOn(scheduler).subscribe());
+
+            const actual: Array<DirectLineExport.Activity> = [];
+            subscriptions.push(directline.activity$.subscribe(a => actual.push(a)));
+
+            scheduler.flush();
+
+            expect(normalActivity.deliveryMode).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (pranavjoshi001) [reply src/directLine.test.ts]: If you leave it undefined then ABS consider it "normal" anyways so I wanted to have that specific scenario where considering existing behavior, i don't pass anything and it should work fine and only passing stream when streaming is on
